csharp/src/Ice/ByteBufferExtensions.cs

```diff
@@ -84,11 +84,22 @@ public static class ByteBufferExtensions
         public static void ReadEmptyEncapsulation(
             this ReadOnlyMemory<byte> buffer,
             Encoding encoding,
-            Communicator communicator) =>
-            new InputStream(buffer,
+            Communicator communicator)
+        {
+            var istr = new InputStream(buffer,
                             encoding,
                             communicator,
-                            startEncapsulation: true).CheckEndOfBuffer(skipTaggedParams: true);
+                            startEncapsulation: true);
+            if (istr.Encoding == Encoding.V2_0)
+            {
+                byte compressionStatus = istr.ReadByte();
+                if (compressionStatus != 0)
```

Comment:
Yes, for ReadEncapsulation otherwise the payload reader will fail otherwise,  for ReadEmptyEncapsulation  we never compres it but I added the check for consistency.

Reply:
Keep in mind that an encaps that the reader believes is empty could actually be large compressed encaps with lots of tagged parameters.

Reply:
I would move this "not compressed" check to the InputStream constructor. I think it's very much like this existing check:
```
Encoding = encapsEncoding;
Encoding.CheckSupported();
```

When constructing an InputStream that starts an encapsulation, if the encapsulation encoding is 2.0, the payload must not be compressed.

Reply:
Fixed as suggested